### PR TITLE
Add support for DHCPv6-PD sub-options

### PR DESCRIPTION
**Pull request by davebarrau**
Thoughts on this? I tried to make it look the same as OptionIANA from an options perspective

- Allow for parsed sub-options in OptIAForPrefixDelegation
- Add initial test

### CODE REVIEW

**Comment by pmazzini** [comment]
Thanks for your contribution! It looks good to me.

The only problem is that this will break router7 build cc @insomniacslk:
https://github.com/rtr7/router7/blob/master/internal/dhcp6/dhcp6.go#L285

**Comment by pmazzini** [comment]
Nice, thanks for fixing it. I would also add a deprecated message to the Options() method so that it can be removed later. Similar to:
https://github.com/insomniacslk/dhcp/blob/master/dhcpv6/types.go#L45

**Comment by davebarrau** [comment]
> Nice, thanks for fixing it. I would also add a deprecated message to the Options() method so that it can be removed later. Similar to:
> https://github.com/insomniacslk/dhcp/blob/master/dhcpv6/types.go#L45

Would Options() be removed, or be changed to returning []Option as part of the deprecation? If the function is removed, there's no easy way to get all options otherwise, unless a differently named function is added as part of this pull request.

**Comment by pmazzini** [comment]
For consistency I think it would be removed and the public field can be used instead, similar to OptionIANA.

File: dhcpv6/option_prefixdelegation.go [hunk 1]
```diff
@@ -12,75 +12,114 @@ type OptIAForPrefixDelegation struct {
 	iaId    [4]byte
 	t1      uint32
 	t2      uint32
-	options []byte
+	options []Option
```
Comment (pmazzini) [inline dhcpv6/option_prefixdelegation.go]: I would use capital letter to make this field public similar to OptIANA.

Comment (davebarrau) [reply dhcpv6/option_prefixdelegation.go]: Ah, like how I originally had it, of course. :)

**Comment by insomniacslk** [comment]
sorry, I was on a long flight and I'm seeing this only now, I'm looking at the code now

**Pull request merged**